src/main/java/Duke.java

```diff
@@ -1,10 +1,203 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws DukeException {
+        printWelcome();
+
+        String input;
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        boolean isRunning = true;
+
+        while (isRunning) {
+            input = in.nextLine();
+            String[] command = input.split(" ", 2);
+
+            switch(command[0]) {
+            case "bye":
+                printBye();
+                isRunning = false;
+                break;
+            case "list":
+                printAllTasks(tasks);
+                break;
+            case "mark":
+                try {
+                    markTaskDone(tasks, command[1]);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println("Please indicate the task number to be marked.");
+                }
+                break;
+            case "unmark":
+                try {
+                    markTaskNotDone(tasks, command[1]);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println("Please indicate the task number to be unmarked.");
+                }
+                break;
+            case "todo":
+                try {
+                    addTodo(tasks, command[1]);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println("OOPS!!! The description of a todo cannot be empty.");
+                    //throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
```

Comment:
Try to remove comments if not required. You can always retrieve them from GitHub if you need them in the future.